Account/FxADeviceRegistrator.swift

```diff
@@ -0,0 +1,125 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Deferred
+import Shared
+
+private let log = Logger.syncLogger
+
+// The current version of the device registration, we use this to re-register
+// devices after we update what we send on device registration.
+private let DeviceRegistrationVersion = 1
+
+public enum FxADeviceRegistrationResult {
+    case RegisteredOrUpdated
```

Comment:
Yeah, I had that originally, but it's messy once we hand off to `registerOrUpdateDevice` since it calls out to multiple error handler helpers and we'd have to pass this information along so they'd know what to return. That code definitely not as straightforward as I'd like it to be (and I threw in a few TODOs since I'm not entirely sure what state we should be in after all the error cases). I'll see if I can clean that up a bit more.